examples/text-generation/run_generation.py

```diff
@@ -679,6 +679,10 @@ def rounder(x):
             def download_book(book_id):
                 import os
 
+                if os.path.isfile(f"/tmp/{book_id}.txt"):
```

Comment:
```suggestion
                import glob
                matches = glob.glob(f"/tmp/{book_id}_*.txt")
                if matches:
                    # File(s) found
                    book_path = matches[0]  # or loop through if needed
```

Shouldn't we use pid when looking for the cached copy of the book?